transport/peerlist.go

```diff
@@ -0,0 +1,15 @@
+package transport
+
+import "context"
+
+//go:generate mockgen -destination=transporttest/peerlist.go -package=transporttest go.uber.org/yarpc/transport PeerList
+
+// PeerList is a collection of Peers.  Outbounds request peers from the ChoosePeer to determine where to send requests
+type PeerList interface {
+	PeerSubscriber
```

Comment:
Why is PeerList a PeerSubscriber. Do all PeerLists have to be PeerSubscribers
or is that just an implementation detail? Specifically, would I ever need to
call the PeerSubscriber functions while consuming a PeerList?